code/game/machinery/floor_light.dm

```diff
@@ -171,3 +173,44 @@ var/list/floor_light_cache = list()
 /obj/machinery/floor_light/cultify()
 	default_light_colour = "#FF0000"
 	update_brightness()
+
+/obj/machinery/floor_light/changing
+	name = "changing Floor light"
+	var/last_color_change
+
+/obj/machinery/floor_light/changing/Initialize(mapload)
+	. = ..()
+	last_color_change = world.time
+
+/obj/machinery/floor_light/changing/process(delta_time)
+	. = ..()
+	update_colour()
+
+/obj/machinery/floor_light/changing/proc/update_colour()
```

Comment:
nit: should probably be `update_color` for consistency

Reply:
Fair